projects/ngrx.io/content/guide/signals/signal-store/testing.md

```diff
@@ -0,0 +1,687 @@
+
+However, in practice, you’ll use the `TestBed because it offers many advantages, such as mocking dependencies and triggering the execution of effects.
+
+Furthermore, crucial features of the SignalStore will not work, if they don't run in an injection context. Examples include the `rxMethod`, `inject` in `withMethods()`, and `withHooks()`.
+
+<div class="alert is-helpful">
+
+**Note:**: Using the TestBed is also the recommendation of the [Angular team](https://github.com/angular/angular/issues/54438#issuecomment-1971813177).
+
+</div>
+
+## Testing the Signal Store
+
+Let's assume, we want to test the following Signal Store:
+
+### Globally provided
+
+_movies.store.ts_
+
+```typescript
+import { signalStore, withState } from '@ngrx/signals';
+
+type Movie = {
+  id: number;
+  name: string;
+};
+
+type State = { movies: Movie[] };
+
+export const MoviesStore = signalStore(
+  withState<State>({
+    movies: [
+      { id: 1, name: 'A New Hope' },
+      { id: 2, name: 'Into Darkness' },
+      { id: 3, name: 'The Lord of the Rings' },
+    ],
+  })
+);
+```
+
+The `TestBed` will instantiate the `MoviesStore`, allowing you to test it immediately.
+
+_movies.store.spec.ts_
+
+```typescript
+import { MoviesStore } from './movies-store';
+import { TestBed } from '@angular/core/testing';
+
+describe('MoviesStore', () => {
+  it('should verify that three movies are available', () => {
+    const store = TestBed.inject(MoviesStore);
+
+    expect(store.movies()).toHaveLength(3);
+  });
+});
+```
+
+### Locally provided
+
+This was possible because the `MoviesStore` is provided globally. For local providers, we have to tweak the test a bit.
+
+_movies.store.ts_
+
+```typescript
+export const MoviesStore = signalStore(
+  withState({
+    movies: [
+      // ... entries
+    ],
+  })
+);
+```
+
+The required addition is that the internal `TestingModule` must provide the `MoviesStore`.
+
+_movies.store.spec.ts_
+
+```typescript
+import { MoviesStore } from './movies.store';
+
+describe('MoviesStore', () => {
+  it('should verify that three movies are available', () => {
+    TestBed.configureTestingModule({
+      providers: [MoviesStore],
+    });
+
+    const store = TestBed.inject(MoviesStore);
+
+    expect(store.movies()).toHaveLength(3);
+  });
+});
+```
+
+### `withComputed`
+
+Testing derived values of `withComputed` is also straightforward.
+
+_movies.store.ts_
+
+```typescript
+export const MoviesStore = signalStore(
+  withState({
+    movies: [
+      // ... entries
+    ],
+  }),
+  withComputed((state) => ({
+    moviesCount: computed(() => state.movies().length),
+  }))
+);
+```
+
+_movies.store.spec.ts_
+
+```typescript
+import { MoviesStore } from './movies.store';
+
+describe('MoviesStore', () => {
+  it('should verify that three movies are available', () => {
+    const store = TestBed.inject(MoviesStore);
+
+    expect(store.moviesCount()).toBe(3);
+  });
+});
+```
+
+### `withMethods`, DI, and asynchronous tasks
```

Comment:
I think this is a great example to add, but I think the learning curve is steep from going from "computed" to "async methods".
What about introducing an extra step that shows how to test "simpler" methods that update state in a synchronous way?
The main reason is that I'm afraid that devs will be scared away with the jest fake timers 😅